### ISSUE TITLE
[Enhancement]: Extend Kafka CRD to configure the jmxOptions for the remaining workloads

### ISSUE THREAD

**Issue by jakubmalek** [strimzi/strimzi-kafka-operator#9949]
### Related problem

Right now Kafka CRD provides a way to configure the `jmxOptons` to enable JMX endpoint with or without authentication on port 9999.
This option is missing however for the remaining workloads like user-operator, topic-operator and cruise-control.
This is not that big of a problem, as its still possible to enable the JMX with the environment variable e.g.:

```yaml
env:
  - name: KAFKA_JMX_OPTS
    value: -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.local.only=false -Dcom.sun.management.jmxremote.port=9999 -Dcom.sun.management.jmxremote.rmi.port=9999 -Djava.rmi.server.hostname=127.0.0.1
```

Nevertheless, if you want enable the authentication, and generate username/password, similarly to what Strimzi does for Kafka and Zookeeper, it requires additional effort.

### Suggested solution

Extend the OpenAPI for Kafka CRD to add reference to the `jmxOptions` configuration with its support in the operator.

### Alternatives

The other alternative is to enable it in the environment variables manually, but as described, it requires some effort to configure authentication with generated secrets.

**Comment by scholzj** [comment]
Just to be clear, this is not about the CRD schema as you suggest -> this needs to be actually implemented in the code because otherwise nobody cares about the what is in the custom resource.

I'm also quite curious about the motivation:
* JMX is badly secured and hard to access
* I don't know about Cruise Control. But in the operators, you do not get anything extra out of JMX. Unlike Kafka, we do not use JMX for metrics. Metrics are provided only as Prometheus metrics. So what do you actually get out of it? What is the expected value?

**Comment by jakubmalek** [comment]
Yes, I phrased it as CRD extension, but of course it means the implementation part as well 😅 
I just thought it wouldn't be so complicated to re-use the existing logic for handling the `jmxOptions`.
Whether the JMX endpoint is a "good idea" or not, its up to debate, but I think for coherency reasons, it would be nice if you could enable it the same way.

And yes, `KAFKA_JMX_OPTS` is the example for the cruise-control, where it's supported for others you might re-use the `STRIMZI_JAVA_OPTS` or `JAVA_TOOL_OPTIONS` so it's still possible, but less nice.

In any case, it's not a pressing problem, but I think it would nice to have it.

**Comment by scholzj** [comment]
> Whether the JMX endpoint is a "good idea" or not, its up to debate, but I think for coherency reasons, it would be nice if you could enable it the same way.

To be honest, these things cost effort to implement initially but also to test and maintain afterward. So I think we should not do this _just because_. But you should explain for each of the components what is the actual value of adding it, and how will you use it.

What I'm trying to say is that the _up to debate_ part is happening while the issue is being triaged. So if you have any other reasons why this should be supported for the other components other than _being coherent_, it would be good to raise them now.

**Comment by jakubmalek** [comment]
In general, I think that JMX endpoint can be quite useful, whenever you need to do a deeper investigation, whether it comes to the heap/off-heap memory usage, CPU profiling, etc.
That said, in most cases the Prometheus metrics are sufficient enough, and JMX is only needed when you need to dig a little bit more deeper, on root cause of the observed problems. Of course I also understand that, in most cases, the observed problems will be reported back to to you instead, given that this workload is owned by you. But I often like to check things myself first.

In any case, for me it's fine to enable the JMX as needed, using the environment variables or end jmxOptions configuration. The main reason I reported it, is that looking at it from the user perspective of a Helm chart, it an be a bit confusing, that in some cases you can enable it using a dedicated configuration supported by the Strimzi and in other you need to use a workaround. 

So to summarize, I can accept the fact that the maintenance costs are not justified in this case, given that there is a not so complicated workaround for it

**Comment by scholzj** [comment]
Discussed on the community call on 18.4.: The JMX access seems to have its own issues:
* Authentication does not work in FIPS environments
* Remote access is not possible due to how JMX works and is configured

We do not think that in that situation we want to add it to other components just for _coherency_. Especially given the additional effort for reviews, maintenance, testing etc. If there would be a reason to extend JMX to other components connector to some specific task or scenario, it can be reconsidered. But it does not seem to be the case here.
